e2e/tests/compat.test.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import firebase from 'firebase/compat/app';
+import 'firebase/compat/auth';
+import 'firebase/compat/analytics';
+import 'firebase/compat/app-check';
+import 'firebase/compat/functions';
+import 'firebase/compat/storage';
+import 'firebase/compat/firestore';
+import 'firebase/compat/messaging';
+import 'firebase/compat/performance';
+import 'firebase/compat/database';
+import { config, testAccount } from '../firebase-config';
+import { expect } from 'chai';
+
+describe('COMPAT', () => {
+  let app: firebase.app.App;
+  before(() => {
+    console.log('FIREBASE VERSION', firebase.SDK_VERSION);
+    app = firebase.initializeApp(config);
+    firebase.setLogLevel('warn');
+  });
+
+  after(async () => {
+    await firebase.auth().signOut();
+    await app.delete();
+  });
+
+  describe('AUTH', async () => {
+    let auth: firebase.auth.Auth;
+    it('init auth', () => {
+      auth = firebase.auth();
+    });
+    it('signInWithEmailAndPassword()', async () => {
+      const cred = await auth.signInWithEmailAndPassword(
+        testAccount.email,
+        testAccount.password
+      );
+      console.log('Logged in with test account', cred.user.email);
+      expect(cred.user.email).to.equal(testAccount.email);
+    });
+  });
+
+  describe('APP CHECK', async () => {
+    // @ts-ignore
```

Comment:
Is `ts-ignore` needed?

Reply:
Oh that was temporary when AppCheck typings were broken. Removed.